compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -327,7 +327,27 @@ impl Context {
             }
             updated_sets.insert(largest_set_index, largest_set);
         }
-        self.value_sets = updated_sets.values().cloned().collect();
+        updated_sets.values().cloned().collect()
+    }
+
+    /// Parallel version of merge_sets
+    /// The sets are merged by chunks, and then the chunks are merged together
+    fn merge_sets_par(sets: &[HashSet<ValueId>]) -> Vec<HashSet<ValueId>> {
+        let mut sets = sets.to_owned();
+        let mut len = sets.len();
+        let mut prev_len = len + 1;
+
+        while len > 1000 && len < prev_len {
+            let chunks: Vec<_> = sets.par_chunks(1000).collect();
+            sets = chunks.par_iter().flat_map(|sets| Self::merge_sets(sets)).collect();
```

Comment:
Do we need to `.collect()` then immediately `.par_iter()`? Or can we iterate from the `par_chunks` directly?
Rayon's docs mention `par_chunks` returns a "Parallel iterator over immutable non-overlapping chunks of a slice" so I think we should be able to call `.par_chunks(1000).flat_map(...)` without collecting